### PR TITLE
Added UAVOnix dialect from documentation.

### PR DESCRIPTION
**Pull request by queue-b**
NOTE: This does not match the published UAVonix.xml file provided by MAVLINK maintainers. That file is quite old.

### CODE REVIEW

**Comment by queue-b** [comment]
@brandon-hass this is in support of ADS-B transponder support.  I added all the Uavionix messages to the UAVionix dialect.

Golint doesn't like the "self" parameter in the method definitions, but that matches the style of the  rest of the package.

**Review by brandon-hass — CHANGES_REQUESTED** [review]
I feel like for the sake of consistency, creating the .xml file to generate the code and then modifying the CRC_EXTRA manually (unless you can get the field names to calculate it correctly, which would be ideal) would make more sense. Thoughts?

**Comment by queue-b** [comment]
If you'd prefer I can definitely do that; assuming that works, you want the enum values in the XML as well?

**Comment by queue-b** [comment]
Although -- if we have to manually modify the XMLCrc after generation, doesn't that break the guarantee of the generated code being correct for subsequent users?

**Comment by queue-b** [comment]
Disregard last comment.  Looks like documentation lists byte offsets so we should be GTG.

**Comment by queue-b** [comment]
@brandon-hass Generated UAVionix dialect from the XML message set. CRCs needed to be manually changed because UAVionix doesn't actually give you the correct XML message name. I'm going to reach out and see if they will provide those.

**Comment by queue-b** [comment]
@brandon-hass Also, it looks like the generator is producing incorrect output for the OWN_SHIP_STATIC message. The packing and unpacking functions seem to be incorrect. Any ideas/can you take a look?

**Comment by brandon-hass** [comment]
@botBoss It looks like uAvionix isn't ordering their fields correctly (or at least in the same way the other dialects do), which is causing the output that you're seeing. I thought it was an issue with the generator at first, but that doesn't seem to be the case.

File: mavlink/definitions/uavionix.xml [hunk 1]
```diff
@@ -0,0 +1,399 @@
+<?xml version='1.0'?>
+<mavlink>
+    <include>common.xml</include>
+    <enums>
+      <enum name="UAVIONIX_SDA_LEVEL">
```
Comment (brandon-hass) [inline mavlink/definitions/uavionix.xml:5]: Since this is part of the uAvionix dialect, would there be a problem dropping the `UAVONIX_` prefix?

Comment (queue-b) [reply mavlink/definitions/uavionix.xml:5]: @brandon-hass I saw the other dialects doing prefixes like that, and the MAVLink common message set enums all start with MAVLink, so I thought that was the standard.

Comment (queue-b) [reply mavlink/definitions/uavionix.xml:5]: Well, OK, looking again at least the ualberta guys did :)

Comment (brandon-hass) [reply mavlink/definitions/uavionix.xml:5]: Ah, I see it in a few places. It just felt super verbose, but obviously doesn't matter.

**Review by brandon-hass — APPROVED** [review]

File: mavlink/definitions/uavionix.xml (hunk 1 above)
Comment (queue-b) [reply mavlink/definitions/uavionix.xml:5]: I agree, it's unfortunate that there aren't any real standards in any of these definition files (or MAVLink as a whole).

**Pull request merged**